plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -669,7 +670,28 @@ public static Map<String, String> createTableProperties(ConnectorTableMetadata t
         if (tableMetadata.getComment().isPresent()) {
             propertiesBuilder.put(TABLE_COMMENT, tableMetadata.getComment().get());
         }
-        return propertiesBuilder.buildOrThrow();
+
+        Map<String, String> baseProperties = propertiesBuilder.buildOrThrow();
+        Map<String, String> extraProperties = IcebergTableProperties.getExtraProperties(tableMetadata.getProperties()).orElseGet(ImmutableMap::of);
+
+        Set<String> illegalExtraProperties = Sets.intersection(
+                ImmutableSet.<String>builder()
+                        .add(TABLE_COMMENT)
+                        .addAll(baseProperties.keySet())
+                        .addAll(SUPPORTED_PROPERTIES)
```

Comment:
As mentioned already, I think we should ensure that there are no scenarios in which the  "native" Iceberg table.

These "native" Iceberg table properties are already listed in `io.trino.plugin.iceberg.IcebergUtil#createTableProperties`

We don't want any of the following properties to be specified in `extra_properties`:

- `write.format.default`
- `format-version`
- `orc.bloom.filter.columns`
- `orc.bloom.filter.fpp`
- `comment`

Pls create a static map containing those property names and intersect them with the `extraProperties` keys for validity.

In your current code, the user may be specify ORC bloom filter properties through `extra_properties` when not setting `orc_bloom_filter_columns`.

Reply:
FYI, `orc.bloom.filter.columns` should be `write.orc.bloom.filter.columns`, same as `orc.bloom.filter.fpp`.
I have a PR for this: https://github.com/trinodb/trino/pull/20433

Reply:
I believe there is some misunderstanding here because I am adhering to the definition in the `HiveMetadata` class and moving it here, since I believe it doesn't make sense to import certain properties from `HiveMetadata`. If it's wrong here, it's also wrong in `HiveMetadata`.

Currently `orc.bloom.filter.columns` and `orc.bloom.filter.fpp` are defined in IcebergMetadata, while comment comes from HiveMetadata.

The comment property is not native to Iceberg, as confirmed by the Iceberg TableProperties link: [Iceberg TableProperties](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/TableProperties.java) and as discussed in this [Iceberg Slack thread](https://apache-iceberg.slack.com/archives/C03LG1D563F/p1706217171665909)

My suggestion is to maintain the TABLE_COMMENT here and create a set including TABLE_COMMENT, ORC_BLOOM_FILTER_COLUMNS, ORC_BLOOM_FILTER_FPP, DEFAULT_FILE_FORMAT, and FORMAT_VERSION, as imported from `org.apache.iceberg.TableProperties`, to address the issue @oneonestar mentioned in another comment.

What do you think?

Reply:
@oneonestar I believe it will make sense to continue after your PR is merged, as it will make dealing with the properties easier.

Reply:
#20433 landed - pls rebase

Reply:
Done @findinpath